src/common/config_cacher.h

```diff
@@ -40,7 +40,8 @@ class md_config_cacher_t : public md_config_obs_t {
   md_config_cacher_t(ConfigProxy& conf,
                      const char* const option_name)
     : conf(conf),
-      option_name(option_name) {
+      keys{option_name, nullptr},
+      option_name(keys[0]) {
```

Comment:
This assumes certain guarantees about the life-time of the passed `option_name`. It's nothing new but, if the `config_cacher_t` get rolled out broadlier, it might be an idea for another clean-up.

Reply:
Yes - that's an unfortunate choice of the existing interface.
I do plan to make a deeper change in 'main', and will probably try to change that.
The simplest solution - just strdup() that name. But I'll try to move to std::string-s if possible.